src/routes/catalog.ts

```diff
@@ -105,7 +105,7 @@ export const providerCatalogHandler = async (req: Request, res: Response) => {
   const selfLinks = generateSelfLinks(req);
   const childLinks = (collections ?? []).map(({ id, title }) => ({
     rel: "child",
-    href: `${self}/collections/${encodeURIComponent(id)}`,
+    href: `${self.replace(/\?.*$/, "")}/collections/${encodeURIComponent(id)}`,
```

Comment:
Looks like we are using `.replace(/\?.*$/, "")` in a few places now, catalog.ts, rootCatalog.ts and browse.ts. We should create a util function `getBaseUrl()` that will return base url.

Reply:
I agree, this change is needed